### PR TITLE
Add BaseDocsGenerate and BaseDocsGenReferences Tests

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

Added these tests as part of the requirements for upgrading to dbt-core 1.2. 

#### BaseDocsGenerate
- Modify second_model.sql to explicitly use alternate schema
- Modify models fixture to use our version of second_model.sql
- Modify verify_catalog from dbt-core to exclude source table verification
- Modify unique_schema fixture to prepend schema with rav-test folder
- Modify project_config_update fixture to prevent the seed being created as a view
- Modify profiles fixture to make root_path the same as schema
- Modify expected_catalog fixture to accommodate dremio data types
- Include test_run_and_generate methods so they use our version of verify_catalog

#### BaseDocsGenReferences
- Modify project_config_update fixture to allow creating a view for seeds. This is because the ephemeral_summary models looks for the seed under database.schema (which is only used for views). 
- Modify unique_schema, profiles, and expected catalog fixtures for same reasons as above

#### Utilities
- Modify base_expected_catalog to look for datalake instead of database 
- Modify expected_references_catalog for the same reason above

#### Other
- Relevant imports
- Remove staging snapshot macro

### Related Issue

https://github.com/dremio/dbt-dremio/issues/43
https://github.com/dremio/dbt-dremio/issues/36

### Additional Reviewers

@jlarue26 
@ArgusLi

### CODE REVIEW

File: tests/functional/adapter/test_basic.py [hunk 1]
```diff
@@ -495,3 +512,200 @@ def dbt_profile_data(
...
+            "model.sql": models__model_sql,
+        }
+
+    # Override this fixture to prepend our schema with rav-test
+    # This ensures the schema works with our datalake
+    @pytest.fixture(scope="class")
+    def unique_schema(self, request, prefix) -> str:
+        test_file = request.module.__name__
+        test_file = test_file.split(".")[-1]
+        unique_schema = f"rav-test.{prefix}_{test_file}"
```
Comment (ravjotbrar) [inline tests/functional/adapter/test_basic.py:572]: rav-test should be constant

File: tests/functional/adapter/test_basic.py [hunk 2]
```diff
@@ -495,3 +512,200 @@ def dbt_profile_data(
...
+        alternate_schema = unique_schema + "_test"
+        return {
+            "asset-paths": ["assets", "invalid-asset-paths"],
+            "vars": {
+                "test_schema": unique_schema,
+                "alternate_schema": alternate_schema,
+            },
+            "seeds": {
+                "quote_columns": True,
+                "+twin_strategy": "clone",
```
Comment (ravjotbrar) [inline tests/functional/adapter/test_basic.py:668]: Could remove as clone is default

File: tests/functional/adapter/test_basic.py [hunk 3]
```diff
@@ -495,3 +512,200 @@ def dbt_profile_data(
...
+            },
+            "models": {
+                "+twin_strategy": "prevent",
```
Comment (ravjotbrar) [inline tests/functional/adapter/test_basic.py:671]: Why is this necessary?

**Review by ravjotbrar — COMMENTED** [review]
Try to avoid code duplication

File: tests/functional/adapter/utils/test_utils.py [hunk 4]
```diff
@@ -104,3 +105,264 @@ def check_relations_equal(adapter, relation_names: List, compare_snapshot_cols=F
...
+    case=None,
+    case_columns=False,
+):
+
+    if case is None:
+
+        def case(x):
+            return x
+
+    col_case = case if case_columns else lambda x: x
```
Comment (ArgusLi) [inline tests/functional/adapter/utils/test_utils.py:132]: Sorry, what's the purpose of lambda x: x?

Comment (ravjotbrar) [reply tests/functional/adapter/utils/test_utils.py:132]: This was written by dbt, so I'm not entirely sure. Based on my limited interaction with this method, I assume it has to do with the case of specific column types.

File: tests/functional/adapter/test_basic.py (hunk 2 above)
Comment (ravjotbrar) [reply tests/functional/adapter/test_basic.py:668]: Done

File: tests/functional/adapter/test_basic.py (hunk 3 above)
Comment (ravjotbrar) [reply tests/functional/adapter/test_basic.py:671]: Removed.

File: tests/functional/adapter/basic/test_base_mat.py [hunk 5]
```diff
@@ -14,13 +14,14 @@
...
+from tests.functional.adapter.utils.test_utils import DATALAKE
 
-
+# Unable to insert variable into docstring, so "rav-test" is hardcoded
 schema_base_yml = """
 version: 2
 sources:
   - name: raw
-    database: "rav-test"
+    database: "rav-test"
```
Comment (ArgusLi) [inline tests/functional/adapter/basic/test_base_mat.py:24]: should this not be DATALAKE as well?

Comment (ravjotbrar) [reply tests/functional/adapter/basic/test_base_mat.py:24]: I tried to insert a variable, but it turns out inserting variables into docstrings isn't trivial, especially when the "{" notation is also used to insert variables from the dbt config.
